citrination_client/models/client.py

```diff
@@ -0,0 +1,95 @@
+
+        :param data_view_id: The ID of the data view to retrieve TSNE from
+        :type data_view_id: int
+        :return: The TSNE analysis
+        :rtype: :class:`Tsne`
+        """
+        analysis = self._data_analysis(data_view_id)
+        projections = analysis['projections']
+        tsne = Tsne()
+        for k, v in projections.items():
+            projection = Projection(
+                    xs=v['x'],
+                    ys=v['y'],
+                    responses=v['label'],
+                    tags=v['inputs'],
+                    uids=v['uid']
+                )
+            tsne.add_projection(k, projection)
+
+        return tsne
+
+    def predict(self, data_view_id, candidates, method="scalar", use_prior=True):
+        """
+        Predict endpoint
+
+        :param data_view_id: The ID of the data view to use for prediction
+        :type data_view_id: str
+        :param candidates: A list of candidates to make predictions on
+        :type candidates: list of dicts
+        :param method: Method for propagating predictions through model
+            graphs
+        :type method: str ("scalar" or "from_distribution")
+        :param use_prior:  Whether to apply prior values implied by the property descriptors
+        :type use_prior: bool
+        :return: The result of the prediction
+        :rtype: :class:`PredictionResult`
+        """
+        body = self._get_predict_body(candidates, method, use_prior)
+        failure_message = "Error while making prediction for data view {}".format(data_view_id)
+        response_dict = self._get_success_json(self._post_json(routes.data_view_predict(data_view_id), data=body, failure_message=failure_message))
+        return _get_prediction_result_from_response(response_dict)
+
+    def _data_analysis(self, data_view_id):
+        """
+        Data analysis endpoint.
+
+        :param data_view_id: The model identifier (id number for data views)
+        :return: dictionary containing information about the data, e.g. dCorr and tsne
+        """
+        failure_message = "Error while retrieving data analysis for data view {}".format(data_view_id)
+        return self._get_success_json(self._get(routes.data_analysis(data_view_id), failure_message=failure_message))
+
+    def _get_predict_body(self, candidates, method="scalar", use_prior=True):
+        if not (method == "scalar" or method == "from_distribution"):
+            raise ValueError("{} method not supported".format(method))
+
+        # If a single candidate is passed, wrap in a list for the user
+        if not isinstance(candidates, list):
+            candidates = [candidates]
+
+        return {
+            "predictionRequest": {
+                "predictionSource": method,
+                "usePrior": use_prior,
+                "candidates": candidates
+                }
+            }
+
+def _get_prediction_result_from_response(response):
+    candidate = response['candidates'][0]
```

Comment:
Why only one candidate?  Predict is supposed to support multiple candidates.

Reply:
Ah gotcha. I didn't know that.

Reply:
👍